### PR TITLE
wholeBodyDynamics: migrate from RateThread to PeriodicThread and add a parameter to set the period

### LINKED ISSUE #128: wholeBodyDynamics: Add a parameter to set the periodicity of the thread instead of hardcoding it to 10 milliseconds
(This is done by calling `RateThead::setRate` or `PeriodicThread::setPeriod`).

### PR DESCRIPTION
**Pull request by GiulioRomualdi**
This should close #128 and #127

### CODE REVIEW

**Comment by GiulioRomualdi** [comment]
I tested with a period of `0.01` and `0.001` in simulation and everything went smooth

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 1]
```diff
@@ -884,6 +886,21 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
...
+	if(!prop.find("devicePeriodInSeconds").isDouble())
+	{
+	    yError() << "wholeBodyDynamics : The devicePeriodInSeconds must be a double";
+	    return false;
+	}
+        this->setPeriod(prop.find("devicePeriodInSeconds").asDouble());
+    }
+    else
+    {
+	yError() << "wholeBodyDynamics : The devicePeriodInSeconds parameter is not found the default one is used. Period: " << this->getPeriod() << " seconds.";
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:900]: If it does not prevent the module to start, I would demote this message to ``yWarning``

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 2]
```diff
@@ -884,6 +886,21 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
     yarp::os::Property prop;
     prop.fromString(config.toString().c_str());
 
+    if (prop.check("devicePeriodInSeconds"))
+    {
+	if(!prop.find("devicePeriodInSeconds").isDouble())
```
Comment (traversaro) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:891]: There is a strange indentation here, could it be space/tabs?

Comment (GiulioRomualdi) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: https://github.com/robotology/whole-body-estimators/pull/130/commits/56f2583a0a89f00dcbd89126238c661c3bbacfc3

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp (hunk 1 above)
Comment (GiulioRomualdi) [reply devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp]: https://github.com/robotology/whole-body-estimators/pull/130/commits/01dd7fdebc6c664b114cfcd299016c1dad281501

**Comment by GiulioRomualdi** [comment]
Hi @traversaro and @S-Dafarra, I updated the code with your suggestions. Let me know if there is something missing

**Comment by HosameldinMohamed** [comment]
Thanks @GiulioRomualdi, can you update the [CHANGELOG](https://github.com/robotology/whole-body-estimators/blob/master/CHANGELOG.md) as well?

File: devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp [hunk 3]
```diff
@@ -884,6 +886,21 @@ bool WholeBodyDynamicsDevice::loadSettingsFromConfig(os::Searchable& config)
...
+        if(!prop.find("devicePeriodInSeconds").isDouble())
+        {
+            yError() << "wholeBodyDynamics : The devicePeriodInSeconds must be a double";
+            return false;
+        }
+        this->setPeriod(prop.find("devicePeriodInSeconds").asDouble());
+    }
+    else
+    {
+        yWarning() << "wholeBodyDynamics : The devicePeriodInSeconds parameter is not found the default one is used. Period: " << this->getPeriod() << " seconds.";
```
Comment (S-Dafarra) [inline devices/wholeBodyDynamics/WholeBodyDynamicsDevice.cpp:900]:
```suggestion
        yWarning() << "wholeBodyDynamics : The devicePeriodInSeconds parameter is not found. The default one is used. Period:" << this->getPeriod() << "seconds.";
```

A super small comment. The operator ``<<`` in the ``yarp`` log stream already adds spaces.

**Review by S-Dafarra — APPROVED** [review]

**Review by traversaro — APPROVED** [review]

**Comment by HosameldinMohamed** [comment]
Thanks all!

**Comment by HosameldinMohamed** [comment]
I aligned `master` with `devel`.

**Pull request merged**